nosnoc/problem.py

```diff
@@ -604,21 +613,30 @@ def create_complementarity_constraints(self, sigma_p, tau, Uk, s_elastic):
         opts = self.opts
         if opts.dcs_mode == DcsMode.PDS:
             for j in range(opts.n_s):
-                lam_j = self.Lambda(stage=j)
+                lam_j = self.Lambda(stage = j)
+                lam_prev_j = self.Lambda(stage=j-1)
                 c_j = self.model.c_pds_fun(self.X_fe()[j])
+                c_n_s = self.model.c_pds_fun(self.X_fe()[opts.n_s-1])
                 self.create_complementarity([lam_j], c_j, sigma_p, tau, s_elastic)
-    
+                self.create_complementarity([lam_prev_j], c_n_s, sigma_p, tau, s_elastic)
+                for jj in range(opts.n_s):
+                    c_jj = self.model.c_pds_fun(self.X_fe()[jj])
+                    lam_n_s = self.Lambda(stage=opts.n_s-1)
+                    self.create_complementarity([lam_n_s], c_jj, sigma_p, tau, s_elastic)
+                    self.create_complementarity([lam_j], c_jj, sigma_p, tau, s_elastic)
+
         # ...rest for other modes...
-        for j in range(opts.n_s):
-            z = self.rk_stage_z(j)
-            stage_comps = self.ocp.g_rk_comp_fun(self.X_fe()[j], Uk, z, self.p, self.model.v_global)
-            a, b = ca.horzsplit(stage_comps)
-            self.create_complementarity([a], b, sigma_p, tau, s_elastic)
+        elif opts.use_fesd:
```

Comment:
These are stage complementarities and they should exist regardless of `use_fesd`.

Reply:
Take a look at this code for an example of what yours should be doing. 
You of course need to replace `Theta` with the evaluation of `c_pds` at `X_fe`. 
You would also need to add the previous FE's `c_pds` complement with `Lambda(stage=k)`
```python
for j in range(opts.n_s):
                # cross comp with previous FE
                self.create_complementarity([self.Theta(stage=j)], self.prev_fe.Lambda(stage=-1), sigma_p, tau, s_elastic)
                for jj in range(opts.n_s):
                    # within fe
                    self.create_complementarity([self.Theta(stage=j)], self.Lambda(stage=jj), sigma_p, tau, s_elastic)
```